pkg/generate/app/sourcelookup.go

```diff
@@ -378,7 +381,7 @@ func StrategyAndSourceForRepository(repo *SourceRepository, image *ImageRef) (*B
 		Binary: repo.binary,
 	}
 
-	if repo.Info() != nil && repo.Info().Dockerfile != nil {
+	if (repo.ignoreRepository || repo.forceAddDockerfile) && repo.Info() != nil && repo.Info().Dockerfile != nil {
```

Comment:
i'm still bothered by this ignoreRepository name+lack of comments (what's the scenario where ignoreRepository is true and forceAddDockerfile is false?  I don't think there is one, so just checking forceAddDockerfile should be sufficient here)